### PR TITLE
OSS portion of wrapper-v2

### PR DESCRIPTION
**Pull request by sgmiller**
Changes to Vault to use the v2 of go-kms-wrapping

### CODE REVIEW

File: vault/seal_autoseal.go [hunk 1]
```diff
@@ -87,8 +87,9 @@ func (d *autoSeal) Finalize(ctx context.Context) error {
 	return d.Access.Finalize(ctx)
 }
 
-func (d *autoSeal) BarrierType() string {
-	return d.Type()
+func (d *autoSeal) BarrierType() wrapping.WrapperType {
+	typ, _ := d.Type(context.Background())
```
Comment (stevendpclark) [inline vault/seal_autoseal.go:91]: Might be nice to change the wrapper.Type api to not return an error? 

A bit worrisome that we are ignoring the error returned possibly in the future, none of the implementations I saw will ever return an error at the moment.

File: internalshared/configutil/encrypt_decrypt_test.go [hunk 2]
```diff
@@ -89,18 +89,24 @@ telemetry {
...
 
-func (r *reversingWrapper) Type() string                     { return "reversing" }
-func (r *reversingWrapper) KeyID() string                    { return "reverser" }
-func (r *reversingWrapper) HMACKeyID() string                { return "" }
-func (r *reversingWrapper) Init(_ context.Context) error     { return nil }
-func (r *reversingWrapper) Finalize(_ context.Context) error { return nil }
-func (r *reversingWrapper) Encrypt(_ context.Context, input []byte, _ []byte) (*wrapping.EncryptedBlobInfo, error) {
-	return &wrapping.EncryptedBlobInfo{
+func (r *reversingWrapper) Type(_ context.Context) (wrapping.WrapperType, error) {
+	return "reverser", nil
```
Comment (kitography) [inline internalshared/configutil/encrypt_decrypt_test.go:93]: Making a comment here since the string has changed, and I want to check that.

**Review by kitography — APPROVED** [review]
LGTM, left a comment on the one string I need to check.

File: vault/seal_autoseal.go (hunk 1 above)
Comment (sgmiller) [reply vault/seal_autoseal.go:91]: Yeah, I didn't love this either, but I don't think we can change wrapping.Type.  Wonder if I can at least log an error here, let me look.

Comment (stevendpclark) [reply vault/seal_autoseal.go]: Addressed by latest commit, thanks Scott.

**Review by stevendpclark — APPROVED** [review]

**Pull request merged**